trieste/models/model_interfaces.py

```diff
@@ -357,7 +368,7 @@ def model(self) -> GPR | SGPR:
         return self._model
 
     def update(self, dataset: Dataset) -> None:
-        x, y = self.model.data
+        x, y = map(lambda var: var.value(), self.model.data)
```

Comment:
what happens if you keep using these as variables?

Reply:
What do you mean?

Reply:
does .value mean that the variable just becomes a constant?

Reply:
I think that value() is needed to get the shape a few lines further down. Also to for the type checking on the DataSet initialisation to pass. Either way, it's fine to be explicit here.